test/services/s3.spec.js

```diff
@@ -2536,11 +2536,61 @@ describe('AWS.S3', function() {
         );
         done();
       });
-
       invocationDeferred = true;
     });
   });
+  if (typeof Promise === 'function') {
+    return describe('getSignedUrlPromise', function() {
+      var catchFunction, resolveFunction, err, url, date;
+      err = null;
+      url = null;
+      date = null;
+
+      catchFunction = function(e) {
+        err = e;
+      };
+
+      resolveFunction = function(u) {
+        url = u;
+      };
+
+      beforeEach(function(done) {
+        err = null;
+        url = null
+        date = AWS.util.date.getDate;
+        AWS.util.addPromises(AWS.S3, Promise);
+        AWS.util.date.getDate = function() {
+          return new Date(0);
+        };
+        return done();
+      });
+
+      afterEach(function(done) {
+        AWS.util.date.getDate = date;
+        done();
+      });
 
+      it('resolves when getSignedUrl is successful', function() {
+        return s3.getSignedUrlPromise('getObject', {
+          Bucket: 'bucket',
+          Key: 'key'
+        }).then(resolveFunction).catch(catchFunction).then(function() {
```

Comment:
Rather than setting the resolved value or error to function-scoped variables, why not just make assertions in the success handler passed to `then` and throw in the error handler passed to `catch`?

The file to which these tests are being added was originally written in CoffeeScript and subsequently replaced with its compiled JavaScript form, so many of the odd conventions used were injected by the CoffeeScript compiler.